### PR TITLE
fix(no-fee november): Relay limits for edge cases for no-fee campaigns

### PR DESCRIPTION
**Pull request by akshay-ap**
## Summary

There is a bug in current backend logic for relay limits on edge cases. This PR fixes the the bug and adds edge test cases.
The env var has been changed to make relay rules more clear to understand.

The bug: If env var config, `balance` is assumed to be whole number. So, if user has balance 99.0000000...1 it is rounded to 99. and same for other cases. The PR changes this and enforces that token balances are in `wei` to avoid any rounding issues.
Also, this pr proposes `balanceMin` and `balanceMax` in relay rule so that its is clear to understand the range for relay limit based on token balance.

The PR adds `RELAY_DAILY_LIMIT_CHAIN_IDS` which will take preference over no-fee campaign. This will help to revert to daily limit relaying without having to make a new release.

## Changes
- Configuration changes
- Change in env var related to relay rules
- Update error message

---

> [!NOTE]
> Moves no-fee relay rules to balance ranges using bigint, prioritizes daily-limit relayer via new chainIds config, and updates validation/tests accordingly.
> 
> - **Config & Schema**
>   - Add `RELAY_DAILY_LIMIT_CHAIN_IDS` env; parsed into `relay.dailyLimitRelayerChainsIds`.
>   - Replace no-fee relay rules with ranged format: `[{ balanceMin: string, balanceMax: string, limit: number }]` in `RELAY_NO_FEE_CAMPAIGN_*_RELAY_RULES`.
>   - Update Zod validator and error messages to require bigint string balances and non-negative, ordered ranges.
> - **Relay Domain**
>   - `RelayRule` now uses `bigint` `balanceMin`/`balanceMax`; `NoFeeCampaignRelayer` compares balances as `bigint` in wei and matches within ranges; returns `0` if no match.
>   - `getTokenBalance` returns raw wei as `bigint`.
>   - `RelayManager` prioritizes `DailyLimitRelayer` when `chainId` is in `relay.dailyLimitRelayerChainsIds`; otherwise falls back to no-fee/daily-limit.
> - **Tests**
>   - Update config validator specs and test configuration data to new ranged rules and messages.
>   - Extend relay controller tests with bigint balances, edge-case thresholds, and gasLimit handling; add scaling helper.
> - **Docs**
>   - `.env.sample` updated with `RELAY_DAILY_LIMIT_CHAIN_IDS` and new no-fee rules examples.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 35a928513c542fc694a006b18c59e788321289be. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/config/entities/configuration.ts [hunk 1]
```diff
@@ -676,9 +678,11 @@ const parseRelayRules = (
     (rule) =>
       typeof rule === 'object' &&
       rule !== null &&
-      typeof rule.balance === 'number' &&
+      typeof rule.balanceMin === 'string' &&
+      typeof rule.balanceMax === 'string' &&
       typeof rule.limit === 'number' &&
-      rule.balance >= 0 &&
+      BigInt(rule.balanceMin) >= 0 &&
+      BigInt(rule.balanceMax) >= BigInt(rule.balanceMin) &&
```
[the remark this answers is not in the document]
Comment (akshay-ap) [reply src/config/entities/configuration.ts]: rules are validated here: https://github.com/safe-global/safe-client-gateway/blob/953278290d30274602e19ad6cdd8fc7a72b077b4/src/config/entities/schemas/configuration.schema.ts#L2-L3

File: src/config/configuration.validator.spec.ts [hunk 2]
```diff
@@ -355,7 +361,7 @@ describe('Configuration validator', () => {
           configurationValidator(config, RootConfigurationSchema),
         ).toThrow(
           new RegExp(
-            `Configuration is invalid: ${fieldKey} Must be a valid JSON array of objects with balance \\(number >= 0\\) and limit \\(number >= 0\\) properties`,
+            `Configuration is invalid: ${fieldKey} Must be a valid JSON array of objects with balances \\(bigint >= 0\\) and limit \\(number >= 0\\) properties`,
```
[the remark this answers is not in the document]
Comment (akshay-ap) [reply src/config/configuration.validator.spec.ts]: https://github.com/safe-global/safe-client-gateway/pull/2773/commits/06444cebcadf2fb544422bd6fa2b08416adfb2b6

File: .env.sample [hunk 3]
```diff
@@ -66,20 +66,21 @@
 # RELAY_PROVIDER_API_KEY_LINEA=
 # RELAY_PROVIDER_API_KEY_BLAST=
 # RELAY_PROVIDER_API_KEY_SEPOLIA=
-
+# The chain ids to apply daily limit relayer feature (comma-separated chainIds).
+# RELAY_DAILY_LIMIT_CHAIN_IDS=11155111
```
[the remark this answers is not in the document]
Comment (gjeanmart) [reply .env.sample]: Nevermind, I think I understood with this commend
> // Prioritize daily limit relayer if chainId is configured for it
